edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/middleware/AuthenticationMiddleware.cs

```diff
@@ -133,12 +133,6 @@ internal async Task<(bool, string)> AuthenticateRequest(HttpContext context)
             IIdentity identity = clientCredentials.Identity;
             IAuthenticator authenticator = await this.authenticatorTask;
 
-            // A downstream module cannot invoke a method on a device by connecting to EdgeHub
-            if (deviceId != this.edgeDeviceId)
```

Comment:
I think this should move to the TwinsController?

Reply:
Filed bug to track refactor the middleware:

[https://msazure.visualstudio.com/One/_workitems/edit/7649811](https://msazure.visualstudio.com/One/_workitems/edit/7649811)

---
In reply to: [451716961](https://github.com/Azure/iotedge/pull/3149#discussion_r451716961) [](ancestors = 451716961)